pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -180,17 +204,41 @@ func (p *PlaceholderPodController) reconcileObjectDeployment(dep *appsv1.Deploym
 		return result, nil
 	}
 
+	// Verify ServiceAccount exists (if Gateway)
```

Comment:
why are we verifying this? seems unnecessary for this logic to need to confirm that it exists. The Pod itself will check that and it's another reconciler that creates the SA

Reply:
second point is no longer valid but would also add that I'd want to ensure we're checking that the SA is enabled with workload identity, which we do here